### DISCUSSION TITLE
State of kubernetes

### DISCUSSION THREAD

**Discussion by xinix00** [ravendb/ravendb#12611] [Q&A]
Last time I used Kubernetes and RavenDB for a customer, there were issues with the using of SWAP, you can't use it inside a kubernetes pod. SWAP is highly recommended and without it can get a OOM-killed which I had a few times. Is this still the case? And can I better use a normall VPS with added SWAP for the RavenDB servers, and Kubernetes for the services that it needs to run?

**Comment by ayende** [comment]
You can use RavenDB inside K8s. 

FYI, you can use Swap on K8s: 
https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2400-node-swap/README.md#design-details

However, in general, while we recommend that you'll have a swap file / partition, it isn't absolutely required if you are provisioning the rest of the system properly.

RavenDB is able to understand the container limits and will respect them and will usually be the only thing on the pod anyway, so it won't be competing with other things that may cause memory issues.
